Generator.Equals/Models/EqualityTypeModel.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+
+namespace Generator.Equals.Models;
+
+internal sealed record class EqualityTypeModel
+{
+    public SyntaxKind SyntaxKind { get; init; }
+
+    public string TypeName { get; }
+    public string? BaseTypeName { get; init; }
+    public bool IsSealed { get; init; }
+    public ImmutableArray<ContainingSymbol> ContainingSymbols { get; }
```

Comment:
Use an array type wrapper. `ImmutableArray` is not equatable.

Reply:
``ImmutableArray<T>`` literally implements ``IEquatable<ImmutableArray<T>>``. Am i missing something?

Reply:
Oh ok i see it now.